dask/dataframe/core.py

```diff
@@ -805,4 +831,28 @@ def get(dsk, keys, get=get_sync, **kwargs):
     return get(dsk2, keys, **kwargs)  # use synchronous scheduler for now
 
 
+def pd_split(df, p, seed=0):
+    """ Split DataFrame into multiple pieces pseudorandomly
+
+    >>> df = pd.DataFrame({'a': [1, 2, 3, 4, 5, 6],
+    ...                    'b': [2, 3, 4, 5, 6, 7]})
+
+    >>> a, b = pd_split(df, [0.5, 0.5], seed=123)  # roughly 50/50 split
+    >>> a
+       a  b
+    1  2  3
+    2  3  4
+    5  6  7
+
+    >>> b
+       a  b
+    0  1  2
+    3  4  5
+    4  5  6
+    """
+    p = list(p)
+    index = pseudorandom(len(df), p, seed)
+    return [df.iloc[index == i] for i in range(len(p))]
```

Comment:
ok sorry

``[ df.take(i) for i in range(len(p)) ]``. Using a boolean selection and iloc is pretty slow compared to ``.take``